### PR TITLE
Remove unused imports

### PR DESCRIPTION
**Pull request by joren485**
##### SUMMARY

This PR removes (almost) all unused imports.

##### ISSUE TYPE

 - Bugfix Pull Request

### CODE REVIEW

File: lib/ansible/module_utils/basic.py
Comment (Akasurde) [inline lib/ansible/module_utils/basic.py]: BOOLEANS is required for backward compatibility, So do not remove this.

File: lib/ansible/module_utils/influxdb.py
Comment (Akasurde) [inline lib/ansible/module_utils/influxdb.py]: This is required.

File: lib/ansible/module_utils/vmware_rest_client.py
Comment (Akasurde) [inline lib/ansible/module_utils/vmware_rest_client.py]: This is required.

File: lib/ansible/modules/cloud/vmware/vmware_guest_facts.py
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_guest_facts.py]: I would recommend - to remove this code block all together.

File: lib/ansible/modules/net_tools/nmcli.py
Comment (Akasurde) [inline lib/ansible/modules/net_tools/nmcli.py]: This is required.

File: lib/ansible/modules/network/aos/_aos_external_router.py
Comment (Akasurde) [inline lib/ansible/modules/network/aos/_aos_external_router.py]: no point in modifying deprecated module.

**Review by Akasurde — CHANGES_REQUESTED** [review]
I think - without testing individual modules or code removing import is dangerous.

**Comment by joren485** [comment]
@Akasurde thanks for your review! I made the changes you requested.
I assumed the CI would tell me if any modules did actually require the imports.
If desired, I can remove all changes of imports in `try` `except` blocks, to lower the risk of removing something that is actually needed.

**Comment by abadger** [comment]
Unfortunately, this is going to be impossible to merge because different people will need to review each one.  Some thoughts:

* You can probably do everything in the bin/ directory in a single PR.  If you want, you can ping me (abadger1999) on IRC when you get that and I'll review it.  If you intend to ping me, you can lump in changes under lib/ansible/cli as well and I'll review those in the same batch.
* Things under contrib/ and modules probably need separate PRs of some sort (for some per file.  For others, per group of files).  This depends on whether the modules are maintained by a group of people or by individual maintainers.  Getting in touch with the maintainers/groups of maintainers is the only way to be sure ahead of time that you're changing things at the right level for review (but you can submit at the most granular level until told otherwise if you really want).
* Things in module_utils are, in general, unsafe to do this to.  The reason is that code which imports the module_util may be depending on finding that symbol in the module_util which imported it.  Removing the "unused" import breaks compatibility.
* pylint has an unused-import check.  You'll want to remove that from the disable list and create a list of the files on which that particular check should not be run (until PRs removing the unused imports are merged).  I believe that the config is here: https://github.com/ansible/ansible/tree/devel/test/sanity/pylint/config  and the files to ignore the tests on should be done here: https://github.com/ansible/ansible/blob/devel/test/sanity/pylint/ignore.txt  but you probably want to talk to mattclay about that on IRC.

I may think of more things, but that's all that's occurring to me while looking at the large block of commits.

**Comment by joren485** [comment]
@abadger thanks for your comment. I will look into splitting this into multiple PRs and talking to the people you suggested.

**Pull request closed without merging**
